### ISSUE TITLE
Pass custom kernel to SVC

### ISSUE THREAD

**Issue by cljord** [cstjean/ScikitLearn.jl#103]
Not sure if I missed this in the docs or if this can be done with ScikitLearnBase, but in Scikit Learn, you can define a custom kernel function very easily and then pass it to the SVC during creation, like this:

```
def my_kernel(X, Y):
    """
    We create a custom kernel:

                 (2  0)
    k(X, Y) = X  (    ) Y.T
                 (0  1)
    """
    M = np.array([[2, 0], [0, 1.0]])
    return np.dot(np.dot(X, M), Y.T)

# we create an instance of SVM and fit out data.
clf = svm.SVC(kernel=my_kernel)
```

(from [here](https://scikit-learn.org/stable/auto_examples/svm/plot_custom_kernel.html))

I haven't been able to figure out how to do this with the Julia package and haven't found anything about it in the docs either. If it isn't possible, this would be a convenient feature (if it is, sorry for opening the issue and would be very thankful if somebody could point me in the right direction).

**Comment by cljord** [comment]
The custom kernel I wanted to pass was just a dot product, so I used dot(X, Y) from the LinearAlgebra package (that's when I opened the issue). I tried it again just now to recreate the error, and using dot(X, Y) throws an error, but weirdly, using X * Y' works fine (I'm assuming these are equivalent, haven't found any way to confirm that except for a bit of testing on my own).

Here's a minimal program showing the error

```
using DataFrames
using RDatasets: dataset
using ScikitLearn
using LinearAlgebra
using ScikitLearn.CrossValidation: train_test_split
@sk_import svm: SVC

iris = dataset("datasets", "iris")

X = convert(Array, select(iris, [:SepalLength, :SepalWidth, :PetalLength, :PetalWidth]))
y = convert(Array, iris[!, :Species])

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.4)

function this_does_not_work(X, Y)
	return dot(X, Y')
end

clf = SVC(kernel=this_does_not_work)

fit!(clf, X_train, y_train)

function this_works(X, Y)
	return X * Y'
end

clf = SVC(kernel=this_works)

fit!(clf, X_train, y_train)
```

Using dot(X, Y') (or any other combination) didn't work either. The error I got was this:

> ERROR: PyError ($(Expr(:escape, :(ccall(#= /Users/cljord/.julia/packages/PyCall/BD546/src/pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))) <class 'IndexError'>
IndexError('tuple index out of range')
  File "/Users/cljord/.julia/conda/3/lib/python3.8/site-packages/sklearn/svm/_base.py", line 226, in fit
    fit(X, y, sample_weight, solver_type, kernel, random_seed=seed)
  File "/Users/cljord/.julia/conda/3/lib/python3.8/site-packages/sklearn/svm/_base.py", line 268, in _dense_fit
    if X.shape[0] != X.shape[1]:
